src/webots/gui/WbRenderingDeviceWindow.cpp

```diff
@@ -135,10 +136,22 @@ WbRenderingDeviceWindow::WbRenderingDeviceWindow(WbRenderingDevice *device) :
 }
 
 WbRenderingDeviceWindow::~WbRenderingDeviceWindow() {
+  if (!mContext)
+    return;
+
+  if (!isVisible())
+    show();  // if the window is not exposed mContext->makeCurrent() doesn't work
+
+  const bool success = mContext->makeCurrent(this);
+  assert(success);
+  if (!success)
+    return;
+
   QOpenGLFunctions_3_3_Core *f = mContext->versionFunctions<QOpenGLFunctions_3_3_Core>();
-  f->glDeleteBuffers(2, (GLuint *)&mVboId);
   f->glDeleteVertexArrays(1, &mVaoId);
-  delete mContext;
+  f->glDeleteBuffers(2, (GLuint *)&mVboId);
+  mInitialized = false;
+  mContext->doneCurrent();
```

Comment:
It seems safe to add it here:
```suggestion
  mContext->doneCurrent();
  delete mContext;
```